src/ui/src/js/controllers/add_command_publishing_blocklist.js

```diff
@@ -0,0 +1,141 @@
+
+AddCommandPublishingBlocklistController.$inject = [
+  '$scope',
+  '$uibModalInstance',
+  '$rootScope',
+  'DTOptionsBuilder',
+  'CommandPublishingBlocklistService',
+  '$state',
+  'commandBlocklist',
+];
+
+/**
+ * systemIndexController - Controller for the system index page.
+ * @param  {Object} $scope         Angular's $scope object.
+ * @param {Object} $uibModalInstance Object for the modal popup window.
+ * @param  {Object} $rootScope     Angular's $rootScope object.
+ * @param  {Object} DTOptionsBuilder
+ * @param  {Object} CommandPublishingBlocklistService command publishing blocklist service object
+ * @param  {Object} $state,     Angular's $state, object.
+ * @param {Object} commandBlocklist
+ */
+export default function AddCommandPublishingBlocklistController(
+    $scope,
+    $uibModalInstance,
+    $rootScope,
+    DTOptionsBuilder,
+    CommandPublishingBlocklistService,
+    $state,
+    commandBlocklist,
+) {
+  $scope.instanceCreated = function(_instance) {
+    $scope.dtInstance = _instance;
+  };
+
+  $scope.dtOptions = DTOptionsBuilder.newOptions()
+      .withOption('autoWidth', false)
+      .withOption(
+          'pageLength', 10,
+      )
+      .withOption('order', [
+        [1, 'asc'],
+        [2, 'asc'],
+        [3, 'asc'],
+      ])
+      .withLightColumnFilter({
+        0: {
+          html: 'input',
+          type: 'checkbox',
+          attr: {
+            'class': 'pull-right',
+            'title': 'Select All',
+            'id': 'select-all-checkbox',
+            'style': 'width: 1%;',
+            'onclick': 'document.getElementById("select-all-hidden-checkbox").click()',
+          },
+        },
+        1: {
+          html: 'input',
+          type: 'text',
+          attr: {class: 'form-inline form-control', title: 'Namespace Filter', style: 'height: 1%'},
+        },
+        2: {
+          html: 'input',
+          type: 'text',
+          attr: {class: 'form-inline form-control', title: 'System Filter', style: 'height: 1%'},
+        },
+        3: {
+          html: 'input',
+          type: 'text',
+          attr: {class: 'form-inline form-control', title: 'Commands Filter', style: 'height: 1%'},
+        },
+      })
+      .withBootstrap();
+
+  $scope.cancelSubmission = function() {
+    $uibModalInstance.dismiss('cancel');
+  };
+
+  $scope.onCommandCheckChange = function(command) {
+    command.isChecked = !command.isChecked;
+    $scope.selectAllValue = false;
+    document.getElementById('select-all-checkbox')['checked'] = false;
+  };
+
+  $scope.submitBlocklistEntries = function() {
```

Comment:
There's currently an opportunity for confusion when no items are selected here, because the UI messaging makes it sound like an entry was created anyway.  I made a comment in #1314 in regards to improving the backend portion of this, but in the UI we could do more to help as well.  Deactivating the submit button if no items are selected would probably be the simplest way to avoid confusion.  Alternatively you could have the alert messaging inform them that no entries were selected and then keep the modal open rather than closing it.